### ISSUE TITLE
tapir integration operation name conflict

### ISSUE THREAD

**Issue by fkowal** [ghostdogpr/caliban#374]
Let me just start by thanking you for this amazing work that you've done.
Caliban is absolutely amazing.

Here's an issue I ran into, and after some exploration, I didn't find a way to resolve it.

Here are two endpoint i described using `tapir`
```
val getOrderEndpoint: Endpoint[(UsernamePassword, String), String, OrderResponse, Nothing] =
    orders.get
      .in("orders" / path[String]("orderId")) // <- orders
      .out(jsonBody[OrderResponse])

  val getOrderListEndpoint: Endpoint[UsernamePassword, String, OrderListResponse, Nothing] =
    orders.get
      .in("orders") // <- orders
      .out(jsonBody[OrderListResponse])
```

as far as I can tell [extractPath](https://github.com/ghostdogpr/caliban/blob/master/interop/tapir/src/main/scala/caliban/interop/tapir/package.scala#L165) is responsible for defining the name of the graphql "field"

As a result the generated schema contains

```
{
              "name" : "orders",
              "description" : null,
              "args" : [
                {
                  "name" : "Authorization",
                  "description" : null,
                  "type" : {
                    "kind" : "NON_NULL",
                    "name" : null,
                    "ofType" : {
                      "kind" : "INPUT_OBJECT",
                      "name" : "UsernamePasswordInput",
                      "ofType" : null
                    }
                  },
                  "defaultValue" : null
                },
                {
                  "name" : "orderId",
                  "description" : null,
                  "type" : {
                    "kind" : "NON_NULL",
                    "name" : null,
                    "ofType" : {
                      "kind" : "SCALAR",
                      "name" : "String",
                      "ofType" : null
                    }
                  },
                  "defaultValue" : null
                }
              ],
              "type" : {
                "kind" : "OBJECT",
                "name" : "OrderResponse",
                "ofType" : null
              },
              "isDeprecated" : false,
              "deprecationReason" : null
            },
            {
              "name" : "orders",
              "description" : null,
              "args" : [
                {
                  "name" : "username",
                  "description" : null,
                  "type" : {
                    "kind" : "NON_NULL",
                    "name" : null,
                    "ofType" : {
                      "kind" : "SCALAR",
                      "name" : "String",
                      "ofType" : null
                    }
                  },
                  "defaultValue" : null
                },
                {
                  "name" : "password",
                  "description" : null,
                  "type" : {
                    "kind" : "SCALAR",
                    "name" : "String",
                    "ofType" : null
                  },
                  "defaultValue" : null
                }
              ],
              "type" : {
                "kind" : "OBJECT",
                "name" : "OrderListResponse",
                "ofType" : null
              },
              "isDeprecated" : false,
              "deprecationReason" : null
            }
```

The problems start with `graphiql.html`
Only one "orders" query is highlighted / available (the second one)

The second issue comes up when I try running the query
```
query($username: String!, $password: String!) {
  orders(username: $username, password: $password) {
    ...
  }
}
```

Caliban responds
```
{
  "data": null,
  "errors": [
    {
      "message": "Argument 'username' is not defined on field 'orders' of type 'Query'."
    }
  ]
}
```

And this probably makes sense because the `username` is not part of the "first" `orders` query
The first "orders" query requires an `Authorization` argument

--------------------------

The solution I would like to propose would be changing the strategy for determining the name of the "query/operation"
Or in general, give the user control to "override" the name

A solution could be be using [EndpointInfo.name](https://github.com/softwaremill/tapir/blob/master/core/src/main/scala/sttp/tapir/Endpoint.scala#L171)

Some pseudo code
```
def expathPath2(endpoint) = 
endpoint.info.name.getOrElse(extractPath(endpoint.input))
```

This would align with the expectations of people using ie. swagger-annotations
where using `@Operation(name = "overrideName")` does the same in the land of openapi

**Comment by ghostdogpr** [comment]
Thanks for the kind words about the library and the detailed explanation.

This is indeed a problem, and in addition to what you said once the WIP PR for https://github.com/ghostdogpr/caliban/issues/228 is merged, the schema validation will fail because each field must be unique according to validation rules.

Will comment on the solution in your PR.

**Comment by ghostdogpr** [comment]
Closed by #375
